File: COADAPT/Entities/Models/Participant.cs
using Microsoft.AspNetCore.Identity;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entities.Models {
	public class Participant : IUserDetails {

		[Key]
		[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		public int Id { get; set; }

		[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		[Column(TypeName = "DATETIME")]
		public DateTime CreatedOn { get; set; }

        public string StressfulEvents { get; set; }

        [Column(TypeName = "DATETIME")]
        public DateTime DateOfBirth { get; set; }

        public string Sex { get; set; }

        public string Education { get; set; }

        public string Children { get; set; }

        public string Parents { get; set; }

        [Column(TypeName = "DATETIME")]
        public DateTime DateOfFirstJob { get; set; }

        [Column(TypeName = "DATETIME")]
        public DateTime DateOfCurrentJob { get; set; }

        public string JobType { get; set; }

        public string MedicalCondition { get; set; }

        public string Substances { get; set; }

        public string PhsychologicalCondition { get; set; }

        public string Region { get; set; }

        [Column(TypeName = "DATETIME")]
        public DateTime StartDate { get; set; }

        [Column(TypeName = "DATETIME")]
        public DateTime EndDate { get; set; }

        [Required]
		public string Code { get; set; }

		public string UserId { get; set; }
		public virtual IdentityUser User { get; set; }

		public int? TherapistId { get; set; }
		public virtual Therapist Therapist { get; set; }

		//public ICollection<SiteParticipant> SiteParticipants { get; set; }
		public ICollection<StudyParticipant> StudyParticipants { get; set; }

	}
}

File: COADAPT-webapp/src/app/models/participant.ts
import { AspNetUser } from './aspnet-user'
import { COADAPTUser } from './coadapt-user'

export interface Participant {
  id: number;
  userName: string;
  password: string;
  user?: AspNetUser;
  code: string;
  // Personal
  dateOfBirth: Date;
  region: string;
  gender: string;
  education: string;
  language: string;
  maritalStatus: string;
  // Job
  jobType: string;
  dateOfFirstJob: Date;
  dateOfCurrentJob: Date;
  // Medical
  startDate: Date;
  endDate: Date;
  // Relational
  siteId: number;
  groupId: number;
  therapistId?: string;
  therapist?: COADAPTUser;
  createdOn: Date;
}
